firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistributionNotificationsManager.java

```diff
@@ -110,6 +110,7 @@ private PendingIntent createAppLaunchIntent() {
       LogWrapper.getInstance().w(TAG + "No activity found to launch app");
       return null;
     }
-    return PendingIntent.getActivity(context, 0, intent, PendingIntent.FLAG_ONE_SHOT);
+    return PendingIntent.getActivity(
+        context, 0, intent, PendingIntent.FLAG_ONE_SHOT | PendingIntent.FLAG_IMMUTABLE);
```

Comment:
This makes me wonder if we should avoid using FLAG_IMMUTABLE on older SDK versions: https://github.com/firebase/firebase-android-sdk/blob/21a9f912862bfbd801690a9f88dce22366bec8ac/firebase-messaging/src/main/java/com/google/firebase/messaging/CommonNotificationBuilder.java#L536-L540

Reply:
Ah, it looks like FLAG_IMMUTABLE was only added in API level 23 (M) https://developer.android.com/reference/android/app/PendingIntent#FLAG_IMMUTABLE

Reply:
The issue is that your sdk has misSdkVersion = 16, so it could crash at runtime?

Reply:
But the check against `Build.VERSION.SDK_INT` would prevent that crash, right @vkryachko ?

Reply:
right, but I don't see the check it in the pr, and I thought that in your previous comment you were suggesting that it's safe not to do the check given that this flag is 23+. If we have the check then it shouldn't be a problem.

Reply:
Ah, no, sorry. I meant we need to add the check. We'll do that now.

Reply:
Done.